Move to stricter CSP

rullzer opened the pull request:
Do not allow unsafe eval by default.


:construction: 

Last merge for https://github.com/orgs/nextcloud/projects/18
This will be rebased several times to validate CSP behavior

TODO:
- [x] ~~This will kill of OC.AddScript~~ converted by mr @ChristophWurst 
  * Need to look into affected apps

rullzer committed:
Disallow unsafe-eval by default

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
Next one: https://github.com/nextcloud/server/blob/0c77b93b8674db71df7d6ca24a8de600818175f6/apps/files/js/filemultiselectmenu.js#L12

rullzer commented:
So I think this is good to get in.

Other stuff we should fix once we run into it.
But if we merge it we can announce it for 15 so app devs can start preparing as well.

ChristophWurst commented on lib/public/AppFramework/Http/ContentSecurityPolicy.php:
```diff
@@ -49,7 +49,7 @@ class ContentSecurityPolicy extends EmptyContentSecurityPolicy {
 	 * TODO: Disallow per default
```
remove

ChristophWurst reviewed (approved):
Let's do this

ChristophWurst commented:
> This will kill of OC.AddScript
> 
>     * Need to look into affected apps

https://github.com/search?q=%22OC.AddScript%22&type=Code have fun :see_no_evil:

ChristophWurst commented:
:boom: 
![bildschirmfoto von 2018-10-09 08-50-24](https://user-images.githubusercontent.com/1374172/46651465-67b63380-cba0-11e8-8fa5-f24a84fd62f2.png)

MorrisJobke commented:
> * Need to look into affected apps

I only have found richdocuments and files_texteditor. 🤷‍♂️

ChristophWurst committed:
Use jQuery.getScript to dynamically load script

Signed-off-by: Christoph Wurst <christoph@winzerhof-wurst.at>

ChristophWurst commented:
> I only have found richdocuments and files_texteditor. man_shrugging

I've pushed my fix that replaces the eval code with `$.getScript`. Could you please test if this actually works for these apps?

rullzer committed:
Fix plural function to be hardcoded

No more weird eval to construct a plural function.
We just use the plural function from symfony.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
@ChristophWurst richdocuments still works with this! yay!

MorrisJobke commented on .htaccess:
```diff
@@ -78,3 +78,7 @@ Options -Indexes
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 /
+ErrorDocument 404 /
```
🙈

MorrisJobke reviewed (approved):
Tested and didn't explode 👍

MorrisJobke commented:
@Arvidas mind to open a new ticket and link here?
